File: dotnet/src/altgraph_shared_app/Services/Graph/v2/Structs/DegreeStruct.cs
using System.Text.Json.Serialization;

namespace altgraph_shared_app.Services.Graph.v2.Structs
{
  public class DegreeStruct
  {
    public long ElapsedMs { get; set; } = -1;
    public string Doctype { get; private set; } = "DegreeStruct";
    [JsonPropertyName("vertex")]
    public string Vertex { get; set; } = string.Empty;
    [JsonPropertyName("degree")]
    public int Degree { get; set; }
    [JsonPropertyName("inDegree")]
    public int InDegree { get; set; }
  }
}

File: java/altgraph_web_app/src/main/java/org/cjoakim/cosmos/altgraph/data/common/graph/v2/struct/DegreeStruct.java
package org.cjoakim.cosmos.altgraph.data.common.graph.v2.struct;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.Data;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties(ignoreUnknown = true)
public class DegreeStruct {

    private long elapsedMs = -1;
    private String doctype = null;
    private String vertex;
    private int degree;
    private int inDegree;

    public DegreeStruct() {
        super();
        setDoctype("DegreeStruct");
    }

    @JsonIgnore
    public String asJson(boolean pretty) throws Exception {

        try {
            ObjectMapper mapper = new ObjectMapper();
            if (pretty) {
                return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(this);
            } else {
                return mapper.writeValueAsString(this);
            }
        } catch (JsonProcessingException e) {
            e.printStackTrace();
            return null;
        }
    }
}
